src/autopas/tuning/tuningStrategy/fuzzyTuning/fuzzyController/FuzzySetFactory.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file FuzzySetFactory.cpp
+ * @author Manuel Lerchner
+ * @date 19.04.24
+ */
+
+#include "FuzzySetFactory.h"
+
+#include <cmath>
+#include <numeric>
+
+#include "autopas/utils/ExceptionHandler.h"
+
+namespace autopas::fuzzy_logic {
+
+std::map<std::string, FuzzySetFactory::AvailableFunctions> FuzzySetFactory::availableFunctionMap = {
+    {"Triangle", FuzzySetFactory::AvailableFunctions::Triangle},
+    {"Trapezoid", FuzzySetFactory::AvailableFunctions::Trapezoid},
+    {"Gaussian", FuzzySetFactory::AvailableFunctions::Gaussian},
+    {"Sigmoid", FuzzySetFactory::AvailableFunctions::Sigmoid},
+    {"SigmoidFinite", FuzzySetFactory::AvailableFunctions::SigmoidFinite}};
```

Comment:
any reason not to make `"Triangle"` etc an enum?

Reply:
I think storing it as a string is needed, as we need to eventually do a string-comparison to match the function stored in the rule-file to the actual implementation.

I haven't found an easy way of converting the enum-values of `FuzzySetFactory::AvailableFunctions` into their string representation without doing some shady macro-stuff or using external libraries such as `magic_enum`, so I figured directly storing the mapping using strings is a better solution. Do you want to change it?

Reply:
Have a look at our `Option` type. If you create your derived option you get enum<->string conversions and more.